bindings/src/Capgemini.PowerApps.SpecFlowBindings/PowerAppsStepDefiner.cs

```diff
@@ -43,18 +43,26 @@ public abstract class PowerAppsStepDefiner
         private static IDictionary<string, string> userProfilesDirectories;
         private static object userProfilesDirectoriesLock = new object();
 
+        /// <summary>
+        /// Date and Time when access token will expire.
+        /// </summary>
+        private static DateTimeOffset accessTokenExpiresOn;
+
         /// <summary>
         /// Gets access token used to authenticate as the application user configured for testing.
         /// </summary>
-        protected static string AccessToken
+        protected static string NewAccessToken
```

Comment:
Might be best to change this to a method (e.g. `GetNewAccessToken()` as it is typically bad practice to have side effects in the getter of a property (other than what you might have during self-initialization). 

That was an issue that was present before your changes, but I'm just noticing it now.

Reply:
That said, renaming it or changing it to a method would effectively be a breaking change as consuming code may be referencing this property.

It might be better to make this an actual property (i.e. with a backing field) and retain the `AccessToken` name. Then we would only have the current code if the backing field is null or the expiry has been reached. That would also fix what is essentially an existing bug where every call to this is generating a new access token. 

With the above, perhaps we could refactor the `authenticatedRecordRepository` to accept the access token as a method parameter rather than a constructor parameter. In hindsight, this would make a lot more sense for something that has a relatively short lifetime.